packages/solid-urql/src/createQuery.ts

```diff
@@ -0,0 +1,337 @@
+): CreateQueryResult => {
+  const client = useClient();
+  const getContext = asAccessor(args.context);
+  const getPause = asAccessor(args.pause);
+  const getRequestPolicy = asAccessor(args.requestPolicy);
+  const getVariables = asAccessor(args.variables);
+
+  const [source, setSource] = createSignal<
+    Source<OperationResult<Data, Variables>> | undefined
+  >(undefined, { equals: false });
+
+  // Combine suspense param coming from context and client with context being priority
+  const isSuspense = createMemo(() => {
+    const ctx = getContext();
+    if (ctx !== undefined && ctx.suspense !== undefined) {
+      return ctx.suspense;
+    }
+
+    return client.suspense;
+  });
+
+  const request = createRequest(args.query, getVariables() as any);
+  const context: Partial<OperationContext> = {
+    requestPolicy: getRequestPolicy(),
+    ...getContext(),
+  };
+  const operation = client.createRequestOperation('query', request, context);
+  const initialResult: CreateQueryState<Data, Variables> = {
+    operation: operation,
+    fetching: false,
+    data: undefined,
+    error: undefined,
+    extensions: undefined,
+    hasNext: false,
+    stale: false,
+  };
+
+  const [result, setResult] =
+    createStore<CreateQueryState<Data, Variables>>(initialResult);
+
+  createComputed(() => {
+    if (getPause() === true) {
+      setSource(undefined);
+      return;
+    }
+
+    const request = createRequest(args.query, getVariables() as any);
+    const context: Partial<OperationContext> = {
+      requestPolicy: getRequestPolicy(),
+      ...getContext(),
+    };
+
+    setSource(() => client.executeQuery(request, context));
+  });
+
+  createComputed(() => {
+    const s = source();
+    if (s === undefined) {
+      setResult(
+        produce(draft => {
+          draft.fetching = false;
+          draft.stale = false;
+        })
+      );
+
+      return;
+    }
+
+    setResult(
+      produce(draft => {
+        draft.fetching = true;
+        draft.stale = false;
+      })
+    );
+
+    onCleanup(
+      pipe(
+        s,
+        onEnd(() => {
+          setResult(
+            produce(draft => {
+              draft.fetching = false;
+              draft.stale = false;
+            })
+          );
+        }),
+        subscribe(res => {
+          setResult(
+            produce(draft => {
+              draft.data = res.data;
+              draft.stale = !!res.stale;
+              draft.fetching = false;
+              draft.error = res.error;
+              draft.operation = res.operation;
+              draft.extensions = res.extensions;
+            })
+          );
+        })
+      ).unsubscribe
+    );
+  });
+
+  const [dataResource, { refetch }] = createResource<
+    CreateQueryState<Data, Variables>,
+    Source<OperationResult<Data, Variables>> | undefined
+  >(source, source => {
+    let sub: Subscription | void;
+    if (source === undefined) {
+      return Promise.resolve(result);
+    }
```

Comment:
I'm not sure whether this comment you left about ordering is accurate as long as the computed effect runs before the resource, which should be the case: https://github.com/urql-graphql/urql/pull/3327#discussion_r1292479692

However, `@urql/vue` does get around this with effect ordering

(See other comment re. effect start)